apps/io-func-sign-user/src/infra/azure/functions/get-qtsp-clauses-metadata.ts

```diff
@@ -10,28 +10,30 @@ import { error, success } from "@internal/io-sign/infra/http/response";
 
 import { QtspClausesMetadataToApiModel } from "../../http/encoders/qtsp-clauses-metadata";
 import { QtspClausesMetadataDetailView } from "../../http/models/QtspClausesMetadataDetailView";
-
-import { NamirialClient } from "../../namirial/client";
-import { makeGetQtspClausesMetadata } from "../../../app/use-cases/get-qts-clauses-metadata";
-
-const makeGetQtspClausesMetadataHandler = (namirialClient: NamirialClient) => {
-  const getQtspClauses = makeGetQtspClausesMetadata(namirialClient);
-
-  const decodeHttpRequest = flow(azure.fromHttpRequest, TE.fromEither);
-
-  const encodeHttpSuccessResponse = flow(
-    QtspClausesMetadataToApiModel.encode,
-    success(QtspClausesMetadataDetailView)
+import { makeGetClausesWithToken, makeGetToken } from "../../namirial/client";
+import { NamirialConfig } from "../../namirial/config";
+import { NamirialClausesToQtspClauses } from "../../http/encoders/namirial-causes-metadata";
+
+const getQtspClausesWithToken = makeGetClausesWithToken()(makeGetToken());
+
+const encodeHttpSuccessResponse = flow(
+  QtspClausesMetadataToApiModel.encode,
+  success(QtspClausesMetadataDetailView)
+);
+
+const decodeHttpRequest = flow(azure.fromHttpRequest, TE.fromEither);
+
+export const makeGetQtspClausesMetadataFunction = (config: NamirialConfig) =>
+  pipe(
+    createHandler(
+      decodeHttpRequest,
+      () =>
+        pipe(
+          getQtspClausesWithToken(config),
+          TE.map(NamirialClausesToQtspClauses.encode)
+        ),
+      error,
```

Comment:
```suggestion
    createHandler(
      decodeHttpRequest,
      () =>
        pipe(
          getQtspClausesWithToken(config),
          TE.map(NamirialClausesToQtspClauses.encode)
        ),
      error,
```

```suggestion
    createHandler(
      decodeHttpRequest,
      () =>
        pipe(
          getQtspClausesWithToken(config),
          TE.mapLeft((e) => new HttpBadRequestError(e.message)),
          TE.map(NamirialClausesToQtspClauses.encode)
        ),
      error,
```
In this way, in the event of an error, we will have a left of type Error which is converted into a 500 internal server error, losing the message. I suggest using HttpBadRequestError